pkg/webhook/resourcedistribution/validating/unifiedresource.go

```diff
@@ -0,0 +1,156 @@
+/*
+Copyright 2021 The Kruise Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package validating
+
+import (
+	"context"
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"k8s.io/kubernetes/pkg/api/legacyscheme"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	//add apps group schemes to legacyscheme
+	_ "k8s.io/kubernetes/pkg/apis/apps/install"
+	//add core group schemes to legacyscheme
+	_ "k8s.io/kubernetes/pkg/apis/core/install"
+)
+
+// This file holds all operations about varying resources in ResourceDistribution process;
+// If you want to add new-type resource, you only need to modify this file.
+
+const (
+	SourceResourceDistributionOfResource = "kruise.io/from-resourcedistribution"
+)
+
+// UnifiedResource abstracts all behaviors of Resource
+type UnifiedResource interface {
```

Comment:
Do not define the UnifiedResource by yourself. You should use the interfaces provided by `k8s.io/apimachinery`, such as `metav1.Object` and `schema.ObjectKind`.

Reply:
Wow!  got it！

Reply:
done